store/rootmulti/store.go

```diff
@@ -576,6 +576,73 @@ func (rs *Store) pruneStores() error {
 	return nil
 }
 
+// PruneHistoryVersions prunes the history versions of each substore based on
+// the pruning options.
+func (rs *Store) PruneHistoryVersions() error {
```

Comment:
Why do we need a new function for this? Isn't there already code in the multi-store that prunes historic versions? I'm already concerned about all the cruft and tech debt within the multi-store re pruning and this just adds to it. Can we perhaps extract logic out of existing functions?

Reply:
the existing `pruneStores` is used to prune the IAVL stores on the fly, while we intended to manually prune the IAVL stores offline. I can try to use the existing `pruneStores`.

Reply:
> - `pruning-interval`: N means to delete old states from disk every Nth block.

I have a question here that the current `pruning-interval` allows at most 1/2 of the old states to be pruned.
Could we change its semantics to `keep old states every Nth block` so we can have more aggressive pruning options, saying prune 9/10 of the states?

Reply:
removed `PruneHistoryVersions` and make use of `PruneStores`